Translated Context.md (#93)
reactjs/de.reactjs.org: content/docs/context.md

permalink: docs/context.html
---

> Context ermöglicht es, Daten durch den Komponentenbaum zu übergeben, ohne die Props in jeder Schicht manuell zu übergeben.

Reviewer: ```suggestion
Context ermöglicht es, Daten durch den Komponentenbaum zu leiten, ohne die Props in jeder Schicht manuell zu übergeben.
```

> In einer typischen React-Applikation werden Daten von oben herab (Elternkomponete zu Kindkomponente) durch Props übergeben. Für bestimmte Arten von Props (z.B. lokale Preferenzen, UI Theme), welche von vielen Komponenten innerhalb der Applikation benötigt werden kann dies sehr umständlich werden. Mit Context ist es möglich, solche Werte zwischen Komponenten zu teilen, ohne diese explizit als Prop durch alle Schichten des Baumes zu übergeben. 

Reviewer: ```suggestion
In einer typischen React-Anwendung werden Daten top-down (Elternkomponente zurKindkomponente) durch Props übergeben. Für bestimmte Arten von Props (z.B. lokale Einstellungen, UI Theme), welche von vielen Komponenten innerhalb der Anwendung benötigt werden kann dies sehr umständlich werden. Mit Hilfe von Context ist es möglich, solche Werte zwischen Komponenten zu teilen, ohne diese explizit als Prop durch alle Schichten des Baumes zu geben. 
```

- [Wann Context verwendet werden soll](#when-to-use-context)
- [Bevor du Context verwendest](#before-you-use-context)
- [API](#api)
  - [React.createContext](#reactcreatecontext)
  - [Context.Provider](#contextprovider)
  - [Class.contextType](#classcontexttype)
  - [Context.Consumer](#contextconsumer)
- [Beispiele](#examples)
  - [Dynamischer Context](#dynamic-context)
  - [Updaten des Contextes von einer genesteten Komponente](#updating-context-from-a-nested-component)

>   - [Konsumieren mehrerer Contexten](#consuming-multiple-contexts)

Reviewer: ```suggestion
  - [Konsumieren mehrerer Contexte](#consuming-multiple-contexts)
```

- [Fallen](#caveats)
- [Legacy API](#legacy-api)

## Wann Context verwendet werden soll {#when-to-use-context}

> Context wurde entwickelt, um Daten zu teilen, welche für einen Baum von React Komponenten als "global" bezeichnet werden können, zum Beispiel der aktelle authentifizierter Nutzer, das Theme oder die bevorzugte Sprache. In dem unten angeführten Codebeispiel zum Beispiel, ziehen wir ein Theme Prop manuell durch den Code, um die Button-Komponente zu stylen:

Reviewer: ```suggestion
Context wurde entwickelt, um Daten zu teilen, welche für einen Baum von React Komponenten als "global" bezeichnet werden können, zum Beispiel der aktuelle authentifizierte Nutzer, das Theme oder die bevorzugte Sprache. In dem unten angeführten Codebeispiel, leiten wir ein Theme Prop manuell durch den Code, um die Button-Komponente zu stylen:
```

`embed:context/motivation-problem.js`

Mit der Verwendung von Context können wir das Übergeben der Props durch Zwischenelemente verhindern:

`embed:context/motivation-solution.js`

## Bevor du Context verwendest {#before-you-use-context}

> Context wird hauptsächlich verwendet, wenn Daten von *vielen* Komponenten in verschiedenen genesteten Schichten erreichbar sein müssen. Benütze es sparsam, weil es das Wiederverwenden von Komponenten schwieriger macht.

Reviewer: ```suggestion
Context wird hauptsächlich verwendet, wenn Daten von *vielen* Komponenten in verschiedenen genesteten Schichten erreichbar sein müssen. Benutze es sparsam, weil es das Wiederverwenden von Komponenten schwieriger macht.
```

  return <PageLayout userLink={userLink} />;
}

// Jetzt haben wir:
<Page user={user} avatarSize={avatarSize} />
// ... rendert ...
<PageLayout userLink={...} />
// ... rendert ...
<NavigationBar userLink={...} />
// ... rendert ...
{props.userLink}
```

Mit dieser Veränderung muss nur die obereste Seitenkomponente über die Verwendung der `user` und `avatarSize` Props der `Link` und `Avatar` Komponenten Bescheid wissen.

> Diese *inversion of control* (engl. für Umkehrung der Steuerung) macht den Code, durch das Reduzieren der Anzahl an überreichten Props  durch die Applikation und durch die Steigerung der Kontrolle der Wurzelkomponenten, in vielen Fällen sauberer. Jedoch ist das nicht in jedem Fall die richtige Entscheidung: durch das Schieben der Komplexität höher in den Baum, werden diese Higher-Level-Komponenten komplizierter und erzwingen die Lower-Level-Komponenten flexibler als erwünscht zu sein.

Reviewer: ```suggestion
Diese *inversion of control* (engl. für Umkehrung der Steuerung) macht den Code, durch das Reduzieren der Anzahl an überreichten Props durch die Anwendung und durch die Steigerung der Kontrolle der Root-Komponenten, in vielen Fällen sauberer. Jedoch ist das nicht in jedem Fall die richtige Entscheidung: durch das Schieben der Komplexität höher in den Baum, werden diese Higher-Level-Komponenten komplizierter und erzwingen die Lower-Level-Komponenten flexibler als erwünscht zu sein.
```

> Du bist nicht auf ein einzelnes Kind für eine Komponente beschränkt. Du kann mehrere Kinder durchgeben, sogar mehrere seperate Slots für Kinder aben, [wie hier dokumentiert wird](/docs/composition-vs-inheritance.html#containment):

Reviewer: ```suggestion
Du bist nicht auf ein einzelnes Kind für eine Komponente beschränkt. Du kannst mehrere Kinder durchgeben, sogar mehrere seperate Slots für Kinder haben, [wie hier dokumentiert wird](/docs/composition-vs-inheritance.html#containment):
```
